DynamicMeter.c

```diff
@@ -0,0 +1,92 @@
+/*
+htop - DynamicMeter.c
+(C) 2021 htop dev team
+(C) 2021 Red Hat, Inc.  All Rights Reserved.
+Released under the GNU GPLv2, see the COPYING file
+in the source distribution for its full text.
+*/
+#include "config.h" // IWYU pragma: keep
+
+#include "DynamicMeter.h"
+
+#include "CRT.h"
+#include "Object.h"
+#include "Platform.h"
+#include "ProcessList.h"
+#include "RichString.h"
+#include "XUtils.h"
+
+static const int DynamicMeter_attributes[] = {
+   DYNAMIC_GRAY,
+   DYNAMIC_RED,
+   DYNAMIC_GREEN,
+   DYNAMIC_BLUE,
+   DYNAMIC_CYAN,
+   DYNAMIC_MAGENTA,
+   DYNAMIC_YELLOW,
+   DYNAMIC_WHITE
+};
+
+Hashtable* DynamicMeters_new(void) {
+   return Platform_dynamicMeters();
+}
+
+typedef struct {
+   unsigned int key;
+   const char* name;
+} DynamicIterator;
+
+static void DynamicMeter_compare(ht_key_t key, void* value, void* data) {
+   const DynamicMeter* meter = (const DynamicMeter*)value;
+   DynamicIterator* iter = (DynamicIterator*)data;
+   if (String_eq(iter->name, meter->name))
+      iter->key = key;
+}
+
+unsigned int DynamicMeter_search(const ProcessList* pl, const char* name) {
+   DynamicIterator iter = { .key = 0, .name = name };
+   Hashtable_foreach(pl->dynamicMeters, DynamicMeter_compare, &iter);
+   return iter.key;
+}
+
+const char* DynamicMeter_lookup(const ProcessList* pl, unsigned int key) {
+   const DynamicMeter* meter = Hashtable_get(pl->dynamicMeters, key);
+   return meter->name;
```

Comment:
Missing NULL check for keys that don't exist?

Reply:
Shouldn't ever happen but added this one to be on the safe side.